src/registry_internals.h

```diff
@@ -3,90 +3,184 @@
 
-    avl_tree registry_urls_root_index;
+    avl_tree registry_urls_root_index; ///< AVL tree of urls
 
-    pthread_mutex_t lock;
+    pthread_mutex_t lock; ///< mutex for synchronizing.
 };
 
+/**
+ * Parse a GUID and re-generated to be always lower case.
+ *
+ * This is used as a protection against the variations of GUIDs.
+ *
+ * @param guid to re-generate
+ * @param result filled after call
+ * @return 0 on success, -1 on error.
+ */
 extern int registry_regenerate_guid(const char *guid, char *result);
 
 #include "registry_url.h"
 #include "registry_machine.h"
 #include "registry_person.h"
 #include "registry.h"
 
-extern struct registry registry;
+extern struct registry registry; ///< Global registry.
 
+/**
+ * Get GUID of this machine.
+ *
+ * @return guid of this machine.
+ */
 extern char *registry_get_this_machine_guid(void);
 
 // REGISTRY LOW-LEVEL REQUESTS (in registry-internals.c)
+
+/**
+ * Access the registry with `person_guid`, `machine_guid`, `url` and `name`.
+ *
+ * @param person_guid of accessing person.
+ * @param machine_guid of accessing machine.
+ * @param url of accessing machine.
+ * @param name of accessing machine.
+ * @param when Now.
+ * @return corresponding REGIStRY_PERSON.
+ */
 extern REGISTRY_PERSON *registry_request_access(char *person_guid, char *machine_guid, char *url, char *name, time_t when);
+/**
+ * Try to delete an URL from REGISTRY_PERSON.
+ *
+ * @param person_guid of accessing person.
+ * @param machine_guid of accessing machine.
+ * @param url of accessing machine.
+ * @param delete_url to delete.
+ * @param when Now.
+ * @return maybe changed REGISTRY_PERSON.
+ */
 extern REGISTRY_PERSON *registry_request_delete(char *person_guid, char *machine_guid, char *url, char *delete_url, time_t when);
+/**
+ * Get REGISTRY_MACHINE for `person_guid`, `machine_guid`, and `url`.
+ *
+ * @param person_guid of accessing person.
+ * @param machine_guid of accessing machine.
+ * @param url of accessing machine.
+ * @param request_machine to gather.
+ * @param when Now.
+ * @return REGISTRY_MACHINE.
+ */
 extern REGISTRY_MACHINE *registry_request_machine(char *person_guid, char *machine_guid, char *url, char *request_machine, time_t when);
 
 // REGISTRY LOG (in registry_log.c)
+
+/**
+ * Log a registry action.
+ *
+ * @param action string
+ * @param p person
+ * @param m machine
+ * @param u url
+ * @param name ktsaou: Your help needed.
+ */
 extern void registry_log(const char action, REGISTRY_PERSON *p, REGISTRY_MACHINE *m, REGISTRY_URL *u, char *name);
+/**
+ * (re)open the log file of the registry.
+ *
+ * @return 0 on success, -1 on error.
+ */
 extern int registry_log_open(void);
+/**
+ * Close the log file of the registry.
+ */
 extern void registry_log_close(void);
+/**
+ * (re)open and truncate the log file of the registry.
+ */
 extern void registry_log_recreate(void);
+/**
+ * ktsaou: Your help needed.
+ *
+ * @return ktsaou: Your help needed.
```

Comment:
I understand that log is something like a DB. Can you please clarify what's the difference between registry_log_load and registry_db_load?